truth_set/java/structure/organisation/move-functionality/UserService.java

```diff
@@ -0,0 +1,48 @@
+import java.util.List;
+import java.util.ArrayList;
+
+class Order {
+    private int id;
+    private String product;
+    private int quantity;
+
+    public Order(int id, String product, int quantity) {
+        this.id = id;
+        this.product = product;
+        this.quantity = quantity;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public String getProduct() {
+        return product;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+}
+
+class OrderService {
+    private List<Order> orders;
+
+    public OrderService() {
+        this.orders = new ArrayList<>();
+    }
+
+    public void placeOrder(Order order) {
+        orders.add(order);
+        notify("Order placed: " + order.getId());
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider separating order and notification logic</b></div>
<div id="fix">

The `placeOrder` method directly manipulates the order list and handles notifications. Consider separating these concerns by extracting the notification logic into a dedicated service.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -31,8 +31,9 @@
  public class OrderService {
      private List<Order> orders;
 +    private NotificationService notificationService;
 
 -    public OrderService() {
 +    public OrderService(NotificationService notificationService) {
          this.orders = new ArrayList<>();
 +        this.notificationService = notificationService;
      }
 
      public void placeOrder(Order order) {
          orders.add(order);
 -        notify("Order placed: " + order.getId());
 +        notificationService.notify("Order placed: " + order.getId());
      }
  }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2557471099>#913ded</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged